includes/Media/Base_Color.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Class Image_Size
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Media;
+
+use Google\Web_Stories\Service_Base;
+use Google\Web_Stories\Infrastructure\HasMeta;
+
+/**
+ * Class Base_Color
+ *
+ * @package Google\Web_Stories\Media
+ */
+class Base_Color extends Service_Base implements HasMeta {
+
+	/**
+	 * The base color meta key.
+	 *
+	 * @var string
+	 */
+	const BASE_COLOR_POST_META_KEY = 'web_stories_base_color';
+
+	/**
+	 * Init.
+	 *
+	 * @since 1.15.0
+	 *
+	 * @return void
+	 */
+	public function register() {
+		$this->register_meta();
+
+		add_filter( 'wp_prepare_attachment_for_js', [ $this, 'wp_prepare_attachment_for_js' ] );
+	}
+
+	/**
+	 * Register meta
+	 *
+	 * @since 1.15.0
+	 *
+	 * @return void
+	 */
+	public function register_meta() {
+		register_meta(
```

Comment:
Why not using `register_post_meta` here btw?

Reply:
We don't use `register_post_meta` in other places. It is all the same to me. register_post_meta is just a wrapper function.

Reply:
We _do_ use it in `Story_Post_Type`.

Reply:
@swissspidy You add this line - https://github.com/google/web-stories-wp/commit/9d18d5f52f1b8a88f170df6be099b0a9d69e40c2

Personally, I like using `register_meta`, it gives you full control, I don't really care either way.